addons/marketing_card/tests/test_campaign.py

```diff
@@ -323,3 +323,56 @@ def test_campaign_ownership(self):
         with self.assertRaises(exceptions.AccessError):
             campaign_as_other.unlink()
         campaign_as_owner.unlink()
+
+    def test_mail_render_security(self):
+        """Asserts the security of the QWeb rendering of the field body_html of card.campaign
+
+        The security is based on the fact:
+        - `body_html` is a related to a `card.template.body`,
+        - `card.template` can only be edited by `base.group_system`,
+        - `body_html` is not `store=True, readonly=False`:
+          The user or manager cannot edit the body html on the campaign directly.
+
+        Hence, only admin can control the `body_html` of the campaign, hence it can be rendered without restriction.
+
+        If one of the above assumptions would change,
+        then the flag `_unrestricted_rendering = True` must be reconsidered.
+        """
+        campaign = self.campaign.with_user(self.marketing_card_manager)
+        arbitrary_qweb = """
+        <img t-attf-src="data:image/png;base64,{{object.env.ref('base.user_admin').sudo().image_128}}"/>
+        """
+
+        campaign.body_html = arbitrary_qweb
+        # Normally, this should raise an AccessError, as `body_html` is a related to `card_template_id.body`
+        # and the user does not have the write rights on `card.template`.
+        # However, the ORM doesn't forward the new value to the related,
+        # and the new value is put in the cache without error.
+        # In the real world, using the web client or the XMLRPC API,
+        # a user would have to do this operation in two requests:
+        # First set the body_html on the campaign,
+        # Then trigger the render (e.g. with `action_update_cards`),
+        # and the cache would have changed between the two operations, hence setting an arbitrary value on body_html
+        # on a campaign wouldn't worK.
+        # Just ensure that the value is well not written in db, nor on the current campaign, nor on the related.
+        # Force a cache invalidation to force a re-fetch from database
+        campaign.invalidate_recordset(fnames=['body_html'])
+        self.assertTrue(arbitrary_qweb not in campaign.body_html)
+        self.assertTrue(arbitrary_qweb not in campaign.card_template_id.body)
+
+        with self.assertRaisesRegex(exceptions.AccessError, 'You are not allowed to modify'):
+            campaign.card_template_id.body = arbitrary_qweb
+
+        # Asserts all render fields are related to card.template, not stored on the campaign itself, and readonly
```

Comment:
Can be moved into a separate check like ``test_render_security_consistency`` or something like that